dask_image/ndinterp/__init__.py

```diff
@@ -0,0 +1,236 @@
+    block_indices = product(*(range(len(bds)) for bds in normalized_chunks))
+    block_offsets = [np.cumsum((0,) + bds[:-1]) for bds in normalized_chunks]
+
+    # use dispatching mechanism to determine backend
+    affine_transform_method = dispatch_affine_transform(image)
+    asarray_method = dispatch_asarray(image)
+
+    # construct dask graph for output array
+    # using unique and deterministic identifier
+    output_name = 'affine_transform-' + tokenize(image, matrix, offset,
+                                                 output_shape, output_chunks,
+                                                 kwargs)
+    output_layer = {}
+    rel_images = []
+    for ib, block_ind in enumerate(block_indices):
+
+        out_chunk_shape = [normalized_chunks[dim][block_ind[dim]]
+                           for dim in range(n)]
+        out_chunk_offset = [block_offsets[dim][block_ind[dim]]
+                            for dim in range(n)]
+
+        out_chunk_edges = np.array([i for i in np.ndindex(tuple([2] * n))])\
+            * np.array(out_chunk_shape) + np.array(out_chunk_offset)
+
+        # map output chunk edges onto input image coordinates
+        # to define the input region relevant for the current chunk
+        if matrix.ndim == 1 and len(matrix) == image.ndim:
+            rel_image_edges = matrix * out_chunk_edges + offset
+        else:
+            rel_image_edges = np.dot(matrix, out_chunk_edges.T).T + offset
+
+        rel_image_i = np.min(rel_image_edges, 0)
+        rel_image_f = np.max(rel_image_edges, 0)
+
+        # Calculate edge coordinates required for the footprint of the
+        # spline kernel according to
+        # https://github.com/scipy/scipy/blob/9c0d08d7d11fc33311a96d2ac3ad73c8f6e3df00/scipy/ndimage/src/ni_interpolation.c#L412-L419 # noqa: E501
+        # Also see this discussion:
+        # https://github.com/dask/dask-image/issues/24#issuecomment-706165593 # noqa: E501
+        for dim in range(n):
+
+            if order % 2 == 0:
+                rel_image_i[dim] += 0.5
+                rel_image_f[dim] += 0.5
+
+            # for some reason the behaviour becomes inconsistent with ndimage
+            # if leaving out the -1 in the next line
+            rel_image_i[dim] = np.floor(rel_image_i[dim]) - order // 2
+            rel_image_f[dim] = np.floor(rel_image_f[dim]) - order // 2 + order
+
+            if order == 0:
+                rel_image_i[dim] -= 1
+
+        # clip image coordinates to image extent
+        for dim, s in zip(range(n), image_shape):
+            rel_image_i[dim] = np.clip(rel_image_i[dim], 0, s - 1)
+            rel_image_f[dim] = np.clip(rel_image_f[dim], 0, s - 1)
+
+        rel_image_slice = tuple([slice(int(rel_image_i[dim]),
+                                       int(rel_image_f[dim]) + 1)
+                                 for dim in range(n)])
+
+        rel_image = image[rel_image_slice]
+
+        # Modify offset to point into cropped image.
```

Comment:
Here lines 196-204 are also commented out. Do these need to be removed?

Reply:
Those intend to explain how `offset` is transformed into `offset_prime` for each output chunk.